src/pages/editor.tsx

```diff
@@ -9,20 +9,64 @@ import {
   CardFooter,
   CardHeader,
   CardTitle,
-} from "~/components/ui/card";
+} from "~/components/ui";
 
 import {
   Accordion,
   AccordionContent,
   AccordionItem,
   AccordionTrigger,
-} from "~/components/ui/accordion";
+} from "~/components/ui";
 
-import { Textarea } from "~/components/ui/textarea";
+import { ScrollArea } from "~/components/ui";
 
-import { Button } from "~/components/ui/button";
+import { Textarea } from "~/components/ui";
+
+import { Button } from "~/components/ui";
+import { AppDispatch, RootState, setEditorText, setResponseText, setVariables, setHeaders, useGraphqlMutation, useAppSelector, useAppDispatch } from "~/rtk";
+import { ChangeEvent, FormEventHandler, MouseEventHandler } from "react";
 
 const Editor: NextPage = () => {
+
+  const [graphql, response] = useGraphqlMutation();
+  const dispatch = useAppDispatch();
+  const data = useAppSelector((state: RootState) => state.data);
```

Comment:
you don't need to specify the state type as you are using correctly typed version here already